core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartbeatDefaultPayload.java

```diff
@@ -59,19 +58,17 @@ public static boolean addDefaultPayLoadProvider(HeartBeatPayloadProviderInterfac
   public static Callable<Boolean> populateDefaultPayload(final List<String> disabledFields, final List<String>
       disabledProviders, final HeartBeatProviderInterface provider) {
     return new Callable<Boolean>() {
-
-      volatile boolean populatedFields = false;
       @Override
       public Boolean call() throws Exception {
+        boolean populatedFields = false;
         for (HeartBeatPayloadProviderInterface payloadProvider : defaultPayloadProviders) {
           if (disabledProviders != null && disabledProviders.contains(payloadProvider.getName())) {
 
             // skip any azure specific modules here
             continue;
           }
 
-          boolean fieldsAreSet = payloadProvider.setDefaultPayload(disabledFields, provider).call();
-          populatedFields = populatedFields || fieldsAreSet;
+          populatedFields |= payloadProvider.setDefaultPayload(disabledFields, provider).call();
```

Comment:
I think this makes it a bit harder to read, since `x |= y` for booleans is easily confused with `x = x || y`. I see you are making very nice usage of the non-short circuiting behavior of `|=`, but it took me a while to parse it correctly in my head 😭.